test/suites/zombie-offchain/test-simple-template-offchain.ts

```diff
@@ -0,0 +1,79 @@
+import { ApiPromise, Keyring } from "@polkadot/api";
+import { signAndSendAndInclude } from "../../util/block.ts";
+describeSuite({
+    id: "ZOF01",
+    title: "Zombie Offchain Tests",
+    foundationMethods: "zombie",
+    testCases: function ({ it, context }) {
+        let relayApi: ApiPromise;
+        let container2000Api: ApiPromise;
+
+        beforeAll(async () => {
+            relayApi = context.polkadotJs("Tanssi-relay");
+            container2000Api = context.polkadotJs("Container2000");
+
+            const relayNetwork = relayApi.consts.system.version.specName.toString();
+            expect(relayNetwork, "Relay API incorrect").to.contain("dancelight");
+
+            const container2000Network = container2000Api.consts.system.version.specName.toString();
+            const paraId2000 = (await container2000Api.query.parachainInfo.parachainId()).toString();
+            expect(container2000Network, "Container2000 API incorrect").to.contain("container-chain-template");
+            expect(paraId2000, "Container2000 API incorrect").to.be.equal("2000");
+
+            // Test block numbers in relay are 0 yet
+            const header2000 = await getHeaderFromRelay(relayApi, 2000);
+
+            expect(header2000.number.toNumber()).to.be.equal(0);
+        }, 50000);
+
+        it({
+            id: "T01",
+            title: "Offchain events are not emitted for simple container chain when offchain pallet testing is not enabled",
+            test: async function () {
+                const blockNum = (await container2000Api.rpc.chain.getBlock()).block.header.number.toNumber();
+                expect(blockNum).to.be.greaterThan(0);
+                await context.waitBlock(10, "Container2000");
+
+                const events = await container2000Api.query.system.events();
+                const offchainWorkerEvents = events.filter((a) => {
+                    return a.event.method == "SimpleOffchainEvent";
+                });
+                expect(offchainWorkerEvents.length).to.be.equal(0);
+            },
+        });
+
+        it({
+            id: "T02",
+            title: "Offchain events are not emitted for simple container chain when offchain pallet testing is enabled",
+            test: async function () {
+                const keyring = new Keyring({ type: "sr25519" });
+                const alice = keyring.addFromUri("//Alice", { name: "Alice default" });
+
+                const blockNum = (await container2000Api.rpc.chain.getBlock()).block.header.number.toNumber();
+                expect(blockNum).to.be.greaterThan(0);
+                const switchTx = container2000Api.tx.offchainWorker.switchOffchainWorker();
+
+                // Enable off-chain worker test event emission
+                await signAndSendAndInclude(container2000Api.tx.sudo.sudo(switchTx), alice);
+                await context.waitBlock(20, "Container2000");
+
+                const events = await container2000Api.query.system.events();
```

Comment:
I believe these events will only be the events of the last block, so even if the offchain worker event is emitted this test may pass. Not sure if we have some util function to find events in the latest N blocks.

Reply:
Yeah good point... I don't think we have a utility function for that but will create one!

Reply:
Added it in https://github.com/moondance-labs/tanssi/pull/723/commits/3faffe08fb84e13a580662ab13bf420dfcfdd40a